src/test/ui/lint/lint-impl-fn.rs

```diff
@@ -29,5 +29,5 @@ fn main() {
 
 #[deny(while_true)]
 fn bar() {
-    while cfg!(unix) {} // no error
+    while cfg!(unix) {} //~ ERROR: infinite loops
```

Comment:
I think this is the relevant part of the diff:
```
--- a/compiler/rustc_lint/src/builtin.rs
+++ b/compiler/rustc_lint/src/builtin.rs
@@ -98,9 +98,9 @@ fn pierce_parens(mut expr: &ast::Expr) -> &ast::Expr {
 impl EarlyLintPass for WhileTrue {
     fn check_expr(&mut self, cx: &EarlyContext<'_>, e: &ast::Expr) {
         if let ast::ExprKind::While(cond, _, label) = &e.kind
-            && let ast::ExprKind::Lit(ref lit) = pierce_parens(cond).kind
-            && let ast::LitKind::Bool(true) = lit.kind
-            && !lit.span.from_expansion()
+            && let ast::ExprKind::Lit(token_lit) = pierce_parens(cond).kind
+            && token_lit.is_bool_true()
+            && !e.span.from_expansion()
         {
             let condition_span = e.span.with_hi(cond.span.hi());
             cx.struct_span_lint(
```
I guess that the `from_expansion()` is now returning false instead of true? Maybe previously the expression span was marked as not-from_expansion but the literal was marked as from-expansion?

Reply:
Oh, I need to use the span from the expression returned by `pierce_parens`. Easy fix.